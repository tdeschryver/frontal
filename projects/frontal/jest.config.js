module.exports = {
  preset: 'jest-preset-angular',
  rootDir: '../../',
  roots: ['<rootDir>/projects/frontal'],
  setupTestFrameworkScriptFile: '<rootDir>/projects/frontal/test.ts',
  globals: {
    'ts-jest': {
      tsConfigFile: './projects/frontal/tsconfig.spec.json',
    },
    __TRANSFORM_HTML__: true,
  },
};
