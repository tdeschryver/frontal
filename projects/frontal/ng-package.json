{
  "$schema": "../../node_modules/ng-packagr/ng-package.schema.json",
  "dest": "../../dist/frontal",
  "deleteDestPath": false,
  "lib": {
    "entryFile": "src/public_api.ts"
  }
}