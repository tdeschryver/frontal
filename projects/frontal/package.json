{
  "name": "frontal",
  "version": "2.0.0",
  "description": "An Angular select/dropdown component",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/timdeschryver/frontal.git"
  },
  "keywords": [
    "Angular",
    "select",
    "selectbox",
    "combobox",
    "typeahead",
    "autocomplete",
    "dropdown"
  ],
  "author": "Tim Deschryver",
  "license": "MIT",
  "bugs": {
    "url": "https://github.com/timdeschryver/frontal/issues"
  },
  "homepage": "https://github.com/timdeschryver/frontal#readme",
  "peerDependencies": {
    "@angular/common": "^6.0.0",
    "@angular/core": "^6.0.0",
    "@angular/forms": "^6.0.0"
  },
  "ng-update": {
    "migrations": "./migrations/migration.json"
   }
}
