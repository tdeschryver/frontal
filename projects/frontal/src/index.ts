export * from './public_api';
export { updateState } from '../src/lib/actions';
