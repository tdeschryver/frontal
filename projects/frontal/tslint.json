{
    "extends": "../../tslint.json",
    "rules": {
        "directive-selector": [
            true,
            "attribute",
            "frontal",
            "camelCase"
        ],
        "component-selector": [
            false,
            "element",
            "frontal",
            "kebab-case"
        ]
    }
}
