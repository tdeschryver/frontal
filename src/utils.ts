let _frontalId = 0;

export function generateId() {
  return `${_frontalId++}`;
}

export function resetId() {
  _frontalId = 0;
}

export const createFrontalInputId = (id: string) => `frontal-input-${id}`;

export const createFrontalButtonId = (id: string) => `frontal-button-${id}`;

export const createFrontalLabelId = (id: string) => `frontal-label-${id}`;

export const createFrontalListId = (id: string) => `frontal-list-${id}`;

export const createFrontalItemId = (frontalId: string, id: string) => `frontal-item-${frontalId}-${id}`;
