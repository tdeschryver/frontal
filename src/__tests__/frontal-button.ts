import { Component } from '@angular/core';
import { TestBed } from '@angular/core/testing';
import { By } from '@angular/platform-browser';
import { FrontalComponent, FrontalButtonDirective } from '../frontal.component';
import { StatusMessagePipe } from '../status.pipe';
import { resetId } from '../utils';

test('generates an id', () => {
  const { button } = setup();
  expect(button.attributes['id']).toBe('frontal-button-0');
});

test('generates an aria labelledby', () => {
  const { button } = setup();
  expect(button.attributes['aria-labelledby']).toBe('frontal-label-0');
});

test('clicking on the button toggles the list', () => {
  const { button, frontal: { state } } = setup();

  expect(state).toMatchObject(expect.objectContaining({ isOpen: false }));

  button.triggerEventHandler('click', {});
  expect(state).toMatchObject(expect.objectContaining({ isOpen: true }));

  button.triggerEventHandler('click', {});
  expect(state).toMatchObject(expect.objectContaining({ isOpen: false }));
});

test('clicking on the button toggles the aria label', () => {
  const { button } = setup();

  expect(button.attributes['aria-label']).toBe('open menu');

  button.triggerEventHandler('click', {});
  expect(button.attributes['aria-label']).toBe('close menu');

  button.triggerEventHandler('click', {});
  expect(button.attributes['aria-label']).toBe('open menu');
});

function setup() {
  resetId();

  TestBed.configureTestingModule({
    declarations: [TestComponent, FrontalComponent, FrontalButtonDirective, StatusMessagePipe],
  });

  TestBed.overrideComponent(TestComponent, {
    set: {
      template: `
        <frontal>
          <ng-template>
            <button frontalButton></button>
          </ng-template>
        </frontal>`,
    },
  });

  const fixture = TestBed.createComponent(TestComponent);
  fixture.detectChanges();

  return {
    fixture,
    frontal: fixture.debugElement.query(By.css('frontal')).componentInstance as FrontalComponent,
    button: fixture.debugElement.query(By.css('[frontalButton]')),
  };
}

@Component({ selector: 'test', template: '' })
class TestComponent {}
