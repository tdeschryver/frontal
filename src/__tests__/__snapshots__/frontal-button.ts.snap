// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sanity check for attributes 1`] = `
<button
  aria-expanded="false"
  aria-haspopup="listbox"
  aria-label="open menu"
  aria-labelledby="frontal-label-0"
  data-toggle="true"
  frontalbutton=""
  id="frontal-button-0"
  role="button"
  type="button"
/>
`;
